From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: softworkz <softworkz@hotmail.com>
Date: Thu, 6 Aug 2026 00:38:45 +0000
Subject: [PATCH] avutil/hwcontext_vulkan: Query export support with the real
 image flags

Query with the flags the images are actually created with. The
DRM-modifier path is unchanged.

Signed-off-by: softworkz <softworkz@hotmail.com>
(cherry picked from commit 63d0237033cca9782932b809f960378d34ee4869)
Signed-off-by: lusoris <lusoris@proton.me>
---
 libavutil/hwcontext_vulkan.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavutil/hwcontext_vulkan.c b/libavutil/hwcontext_vulkan.c
index 940a35e583a5fd86e8b43b1c03af4ff72949fc6f..f1c55773fec03fb62ff49329bdba288d6b57ffd4 100644
--- a/libavutil/hwcontext_vulkan.c
+++ b/libavutil/hwcontext_vulkan.c
@@ -2830,7 +2830,7 @@ static void try_export_flags(AVHWFramesContext *hwfc,
         .tiling = hwctx->tiling,
         .usage  = hwctx->usage,
         .flags  = (hwctx->tiling == VK_IMAGE_TILING_DRM_FORMAT_MODIFIER_EXT && has_mods) ?
-                  (hwctx->img_flags) : (VkImageCreateFlags)(VK_IMAGE_CREATE_ALIAS_BIT),
+                  (hwctx->img_flags) : hwctx->img_flags ? hwctx->img_flags : (VkImageCreateFlags)(VK_IMAGE_CREATE_ALIAS_BIT),
     };
 
     /* The query must describe the image as it will be created: a
